extensions/wikia/Chat2/ChatBanListSpecial_helper.php

```diff
@@ -0,0 +1,243 @@
+			}
+
+			/* number of records */
+			$oRow = $dbs->selectRow(
+				$this->mTable,
+				[ 'count(0) as cnt' ],
+				$where,
+				__METHOD__
+			);
+			if ( is_object( $oRow ) ) {
+				$data['cnt'] = $oRow->cnt;
+			}
+
+			if ( $data['cnt'] > 0 ) {
+				/* select records */
+				$oRes = $dbs->select(
+					[
+						$this->mTable,
+					],
+					[
+						'start_date',
+						'cbu_user_id',
+						'end_date',
+						'cbu_admin_user_id',
+						'reason',
+					],
+					$where,
+					__METHOD__,
+					[
+						'ORDER BY' => $orderby,
+						'LIMIT'    => $this->mLimit,
+						'OFFSET'   => intval( $this->mOffset ),
+					]
+				);
+
+				$data['data'] = [];
+				while ( $oRow = $dbs->fetchObject( $oRes ) ) {
+
+					$oUser = User::newFromId( $oRow->cbu_user_id );
+					$oAdmin = User::newFromId( $oRow->cbu_admin_user_id );
+
+					$data['data'][] = [
+						'timestamp'    => $wgLang->timeanddate( $oRow->start_date, true ),
+						'user'         => $this->sk->link( $oUser->getUserPage(), $oUser->getName() ),
+						'user_actions' => $this->getUserLinks( $oUser ),
+						'expires'      => $wgLang->formatExpiry( $oRow->end_date, true ),
+						'admin_user'   => $this->sk->link( $oAdmin->getUserPage(), $oAdmin->getName() ),
+						'admin_links'  => $this->getUserLinks( $oAdmin ),
+						'reason'       => Linker::commentBlock( $oRow->reason ),
+					];
+
+				}
+				$dbs->freeResult( $oRes );
+
+				if ( !empty( $data ) ) {
+					$wgMemc->set( $memkey, $data, 60 * 60 );
+				}
+			}
+		} else {
+			$data = $cached;
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return $data;
+	}
+
+	private function getUserLinks( $oUser ) {
+		global $wgLang, $wgUser;
+
+		$userIsBlocked = $wgUser->isBlocked( true, false );
+		$this->sk = RequestContext::getMain()->getSkin();
+		$oEncUserName = urlencode( $oUser->getName() );
+		$links = [
+			0 => "",
+			1 => $this->sk->link(
+				Title::newFromText( 'Contributions', NS_SPECIAL ),
+				$wgLang->ucfirst( wfMsg( 'contribslink' ) ),
+				"target={$oEncUserName}"
+			),
+		];
+
+		global $wgEnableWallExt;
+		if ( !empty( $wgEnableWallExt ) ) {
+			$oUTitle = Title::newFromText( $oUser->getName(), NS_USER_WALL );
+			$msg = 'wall-message-wall-shorten';
+		} else {
+			$oUTitle = Title::newFromText( $oUser->getName(), NS_USER_TALK );
+			$msg = 'talkpagelinktext';
+		}
+
+		if ( $oUTitle instanceof Title ) {
+			$links[0] = $this->sk->link( $oUTitle, $wgLang->ucfirst( wfMsg( $msg ) ) );
+		}
+
+		if ( $wgUser->isAllowed( 'block' ) && ( !$userIsBlocked ) ) {
+			$links[] = $this->sk->link(
+				Title::newFromText( "BlockIP/{$oUser->getName()}", NS_SPECIAL ),
+				$wgLang->ucfirst( wfMsg( 'blocklink' ) )
+			);
+		}
+		if ( $wgUser->isAllowed( 'userrights' ) && ( !$userIsBlocked ) ) {
+			$links[] = $this->sk->link(
+				Title::newFromText( 'UserRights', NS_SPECIAL ),
+				$wgLang->ucfirst( wfMsg( 'listgrouprights-rights' ) ),
+				"user={$oEncUserName}"
+			);
+		};
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1116)